### DISCUSSION TITLE
Is extension a must?

### DISCUSSION THREAD

**Discussion by Tomporino** [darkreader/darkreader#7760] [Q&A] [answered]
Hi! I have a problem, which maybe the expected behavior, but I'm not sure so I ask it here before opening an issue. So the situation is that I created an app in Angular, and I installed the darkreader npm package (v 4.9.43) and It works fine untill I opened once while my DarkReader extension on Chrome ( 96.0.4664.110 (Official Build)) was disabled and it starts to throw error mesages (ERROR TypeError: Cannot read properties of null (reading 'disableStyleSheetsProxy')). So I use this package because the people (in theory) maybe don't use the DarkReader browser extension, and I write my code using it, and when I disabled it I see that it's disgusting without it. But when I have my extension on, the site is "double DarkReader" it, but when the extension is not installed (or disabled) it simply not working. The dependency of an installed DarkReader in the browser seems strange for me as if everyone have it I don't have to use the npm package. 
So the main question is: Must I have an extension (Enabled) in the browser to use DarkReader npm package?
I can open a more detailed issue if it's really a problem and not expected behavior. 
(sorry for bad english)

**Comment by Gusted** [comment, marked as answer]
Hi @Tomporino!

TL;DR the extension isn't a must.
The error you mentioned was a oversight that I had and is fixed in the next release(I thought it was already fixed with the current release): https://github.com/darkreader/darkreader/commit/011f099c22e8cc3c1cbf32f92eb84f808eb2203f

Dark reader has built-in code to avoid conflicts between the package and the extension.

**Reply by Tomporino** [reply]
Oh I see, thank you!
